app/src/streamlit_app.py

```diff
@@ -1,29 +1,117 @@
 import streamlit as st
+from llama_stack_client.lib.agents.agent import Agent
+from llama_stack_client import LlamaStackClient, APIConnectionError
+from dotenv import load_dotenv
+import os
+import uuid
+import json
+
+load_dotenv()
+
+BASE_URL = os.getenv("REMOTE_BASE_URL")
+TOOL_DEBUG = False
+
+client = LlamaStackClient(base_url=BASE_URL)
+
+try:
+    models = client.models.list()
+    connected = True
+except APIConnectionError:
+    models = []
+    connected = False
+model_list = [model.identifier for model in 
+              models if model.api_model_type == "llm"]
+
+tool_groups = client.toolgroups.list()
+tool_groups_list = [tool_group.identifier for tool_group in 
+                    tool_groups if tool_group.identifier.startswith("mcp::")]
+
+def reset_agent():
+    st.session_state.clear()
+    st.cache_resource.clear()
 
 st.title("Llama Stack + MCP Client")
 
 with st.sidebar:
+    if connected:
+        st.markdown(":green[Connected]")
+    else:
+        st.markdown(":red[No Connection]")
+    
+    st.header("Model")
+    model = st.selectbox(label="models", options=model_list,index=3, on_change=reset_agent)
+    
     st.header("MCP Servers")
-    tool_groups = ["Ansible","OpenShift","GitHub"]
-    toolgroup_selection = st.pills(label="Available Servers",options=tool_groups, selection_mode="multi",)
-
-    tool_list = ["tool_1","tool_2","tool_3"]
+    toolgroup_selection = st.pills(label="Available Servers",options=tool_groups_list, selection_mode="multi",on_change=reset_agent)    
     
     active_tool_list = []
     for toolgroup_id in toolgroup_selection:
-        active_tool_list.extend([f"{toolgroup_id}_{tool}" for tool in tool_list])
-    st.markdown("Active Tools: ")
+        active_tool_list.extend([t.identifier for t in client.tools.list(toolgroup_id=toolgroup_id)])
+    
+    st.markdown(f"Active Tools: 🛠 {len(active_tool_list)}")
     st.json(active_tool_list)
 
     st.text_input(label="Install New Server", placeholder="MCP Server")
 
+@st.cache_resource
+def create_agent():
+    return Agent(client,
+              model=model,
+              instructions="You are a helpful assistant. When you use a tool always respond with a summary of the result.",
+              tools=toolgroup_selection,
+              sampling_params={"max_tokens":1024},
```

Comment:
is this a default behavior we would like to enable for all prompts being used with different mcp servers to account for the ongoing `vllm out of tokens` issue?

Reply:
actually nvm, yeah. setting it to 1024 seems to avoid the out of tokens error for some of our examples. Will leave it as is for now.